### ISSUE TITLE
[feature request]Add an env variable to test code in different path when using openmp

### ISSUE THREAD

**Issue by MlWoo** [pytorch/pytorch#5096]
Refer to #4824  and #4188 .
In consideration of openmp overhead threshold, actually the code has 2 pathes need to be tested.  However, the size of tensor in test case are usually small in order to increase test speed. So the current code is potentially dangerous because the path with multi-thread maybe is never tested. It's necessary to add an env variable to control all code paths to cover these case.  I think we can start with TH and THNN modules which intensively use openmp. I hope that guys which are in charge of test case could also follow it once the env variable is ready.

**Comment by MlWoo** [comment]
Taking #4824 as an example., in init.cpp of the THNN module, the machine will try to get the env variables from the code below:
```c
int THNN_FORCE_OPENMP = 0;
char* evar = getenv("THNN_FORCE_OPENMP");
if (evar != NULL && strncmp(evar, "1", 2) == 0)
    THNN_FORCE_OPENMP = 1;
```
In VolumetricConvolutionMM.c file, the machine will use the variable like that:
```c
#ifdef _OPENMP
    int use_openmp = (THNN_FORCE_OPENMP) || (T > CONV3D_OMP_THRESHOLD);
    #pragma omp parallel for if (use_openmp) private(t)
#endif
```

**Comment by zou3519** [comment]
I don't think we have something like this yet. It sounds like a worthwhile thing to test.

**Comment by mruberry** [comment]
Increasing test coverage is always good, and we are trying to be more mindful about testing different sizes to trigger new codepaths. 

Triggering an environment variable on/off is also interesting if it exercises an alternative code path. Maybe we could add a decorator for it?

**Comment by VitalyFedyunin** [comment]
I also recommend porting TH functions to Aten instead of wasting time to maintain old code.
